server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/methods/AttachmentChecker.java

```diff
@@ -63,14 +62,17 @@ public AttachmentChecker(AttachmentManager attachmentManager) {
 
     private List<BlobId> listAttachmentsNotFound(List<Attachment> attachments, MailboxSession session) throws MailboxException {
         ThrowingPredicate<Attachment> notExists =
-            attachment -> !attachmentManager.exists(getAttachmentId(attachment), session);
+            attachment -> {
+                try {
+                    blobManager.retrieve(attachment.getBlobId(), session);
```

Comment:
IMO, Retrieving blobs for checking they exist is quite expensive. Should we have a new method for that?

Reply:
Solved in https://github.com/apache/james-project/pull/584

One refactoring at a time!